server/src/main/java/io/druid/server/QueryResourceV3.java

```
@@ -0,0 +1,161 @@
+public class QueryResourceV3 extends QueryResource
+{
+  private final ServiceEmitter emitter;
+  private final ObjectMapper jsonMapper;
+  private final RequestLogger requestLogger;
+
+  @Inject
+  public QueryResourceV3(
+      QueryToolChestWarehouse warehouse,
+      ServerConfig config,
+      @Json ObjectMapper jsonMapper,
+      @Smile ObjectMapper smileMapper,
+      QuerySegmentWalker texasRanger,
+      ServiceEmitter emitter,
+      RequestLogger requestLogger,
+      QueryManager queryManager,
+      AuthConfig authConfig
+  )
+  {
+    super(warehouse, config, jsonMapper, smileMapper, texasRanger, emitter, requestLogger, queryManager, authConfig);
+    this.emitter = emitter;
+    this.jsonMapper = jsonMapper;
+    this.requestLogger = requestLogger;
+  }
+
+  @Override
+  protected Response buildQueryResponse(
+      final HttpServletRequest req,
+      final Query query,
+      final QueryToolChest toolChest,
+      final ObjectWriter jsonWriter,
+      final String contentType,
+      final long startTime,
+      final Yielder yielder,
+      final Map<String, Object> responseContext
+  )
+      throws IOException
+  {
+    try {
+      return Response
+          .ok(
+              new StreamingOutput()
+              {
+                @Override
+                public void write(OutputStream outputStream) throws IOException, WebApplicationException
+                {
+                  CountingOutputStream os = new CountingOutputStream(outputStream);
+
+                  //jsonSerializer would close the yielder.
+                  try {
+                    jsonWriter.writeValue(
+                        os,
+                        ImmutableMap.of(
+                            "result", yielder,
+                            "context", responseContext
+                        )
+                    );
+                  } finally {
+                    os.flush(); // Some types of OutputStream suppress flush errors in the .close() method.
```

I don't think we care as much if flush errors are swallowed on an error somewhere in the jackson workflow.